src/Gui/CMakeLists.txt

```
@@ -78,79 +78,70 @@ else(MSVC)
     )
 endif(MSVC)
 
-if (BUILD_QT5)
+if (QtWebEngineWidgets_FOUND AND BUILD_WITH_CONDA)
+    add_definitions(-DQTWEBENGINE)
+endif()
 
-    if (Qt5WebEngineWidgets_FOUND AND BUILD_WITH_CONDA)
-        add_definitions(-DQTWEBENGINE)
-    endif()
+include_directories(
+    ${QtCore_INCLUDE_DIRS}
+    ${QtWidgets_INCLUDE_DIRS}
+    ${QtOpenGL_INCLUDE_DIRS}
+    ${QtPrintSupport_INCLUDE_DIRS}
+    ${QtSvg_INCLUDE_DIRS}
+    ${QtNetwork_INCLUDE_DIRS}
+    ${QtUiTools_INCLUDE_DIRS}
+)
+list(APPEND FreeCADGui_LIBS
+    ${QtCore_LIBRARIES}
+    ${QtWidgets_LIBRARIES}
+    ${QtOpenGL_LIBRARIES}
+    ${QtPrintSupport_LIBRARIES}
+    ${QtSvg_LIBRARIES}
+    ${QtNetwork_LIBRARIES}
+    ${QtUiTools_LIBRARIES}
+)
 
+if(${Qt5WinExtras_FOUND})
     include_directories(
-        ${Qt5Core_INCLUDE_DIRS}
-        ${Qt5Widgets_INCLUDE_DIRS}
-        ${Qt5OpenGL_INCLUDE_DIRS}
-        ${Qt5PrintSupport_INCLUDE_DIRS}
-        ${Qt5Svg_INCLUDE_DIRS}
-        ${Qt5Network_INCLUDE_DIRS}
-        ${Qt5UiTools_INCLUDE_DIRS}
-    )
-    list(APPEND FreeCADGui_LIBS
-        ${Qt5Core_LIBRARIES}
-        ${Qt5Widgets_LIBRARIES}
-        ${Qt5OpenGL_LIBRARIES}
-        ${Qt5PrintSupport_LIBRARIES}
-        ${Qt5Svg_LIBRARIES}
-        ${Qt5Network_LIBRARIES}
-        ${Qt5UiTools_LIBRARIES}
-    )
-    if(${Qt5WinExtras_FOUND})
-      include_directories(
         ${Qt5WinExtras_INCLUDE_DIRS}
-      )
-      list(APPEND FreeCADGui_LIBS
-        ${Qt5WinExtras_LIBRARIES}
-      )
-    endif()
-else()
-    include_directories(
-        ${QT_INCLUDE_DIR}
     )
     list(APPEND FreeCADGui_LIBS
-        ${QT_LIBRARIES}
-        ${QT_QTUITOOLS_LIBRARY}
+        ${Qt5WinExtras_LIBRARIES}
     )
 endif()
 
 IF(SPNAV_FOUND)
     if(SPNAV_USE_X11)
-      add_definitions(-DSPNAV_USE_X11)
-      if (BUILD_QT5 AND UNIX AND NOT APPLE)
-        find_package(Qt5X11Extras REQUIRED)
-        include_directories(${Qt5X11Extras_INCLUDE_DIRS})
-        list(APPEND FreeCADGui_LIBS ${Qt5X11Extras_LIBRARIES})
-      endif()
-      find_package(X11 QUIET)
-      if (X11_FOUND)
-        list(APPEND FreeCADGui_LIBS
-            ${X11_X11_LIB}
-        )
-      endif(X11_FOUND)
-      SET(FreeCADGui_SDK_SRCS
-        3Dconnexion/GuiAbstractNativeEvent.cpp
-        3Dconnexion/GuiNativeEventLinuxX11.cpp
-        )
-      SET(FreeCADGui_SDK_MOC_HDRS
-        3Dconnexion/GuiAbstractNativeEvent.h
-        3Dconnexion/GuiNativeEventLinuxX11.h
-        )
+        add_definitions(-DSPNAV_USE_X11)
+        if (FREECAD_QT_MAJOR_VERSION EQUAL 5 AND UNIX AND NOT APPLE)
+            find_package(Qt5X11Extras REQUIRED)
+            include_directories(${Qt5X11Extras_INCLUDE_DIRS})
+            list(APPEND FreeCADGui_LIBS ${Qt5X11Extras_LIBRARIES})
+            # Note that Qt6 has removed the QtX11Extras files
```

Does `libspnav` already support Qt6? AFAIK not. So we should probably disable SpaceBall, if we use Qt6?

```suggestion
          elseif(FREECAD_QT_MAJOR_VERSION EQUAL 6 AND UNIX AND NOT APPLE)
            # We don't have an option for SpaceBall but always depend on it, so just set the FOUND to false.
            set(SPNAV_FOUND FALSE CACHE String "SpaceBall doesn't work with Qt6 yet.")
            message(WARNING "SpaceBall does currently not work with Qt6. Disabling SpaceBall!")
```

---

That's the sort of thing we'll take care of in the next stage of Qt6 integration (after all technically _nothing_ in FreeCAD supports Qt6 right now... 😆).

---

JFYI, Upstream libspnav dev has not registered any bugs related to Qt6 yet.

---

Note: line 123 of the revision grammar could be improved to
"SpaceBall **currently does** not work with Qt6."